pkg/storage/types/pvc.go

```diff
@@ -256,3 +256,37 @@ func RenderPVC(size *resource.Quantity, claimName, namespace, storageClass, acce
 
 	return pvc
 }
+
+func IsHotplugVolume(vol *virtv1.Volume) bool {
+	if vol == nil {
+		return false
+	}
+	volSrc := vol.VolumeSource
+	if volSrc.PersistentVolumeClaim != nil && volSrc.PersistentVolumeClaim.Hotpluggable {
+		return true
+	}
+	if volSrc.DataVolume != nil && volSrc.DataVolume.Hotpluggable {
+		return true
+	}
+	if volSrc.MemoryDump != nil && volSrc.MemoryDump.PersistentVolumeClaimVolumeSource.Hotpluggable {
+		return true
+	}
+
+	return false
+}
+
+func GetVolumesByName(vmiSpec *virtv1.VirtualMachineInstanceSpec) map[string]*virtv1.Volume {
+	volumes := map[string]*virtv1.Volume{}
+	for _, vol := range vmiSpec.Volumes {
+		volumes[vol.Name] = vol.DeepCopy()
```

Comment:
Is this necessary?

Reply:
Don't think so in this case but I think it's not a bad idea in theory.  Especially if these functions will be more greatly used/adopted.  Personally never had an additional deepCopy cause an issue but NOT deep copying has caused me much grief

Reply:
@xpivarc what is the concern here? That it would use too much memory. Would you rather prefer a shallow copy? I used a deep copy because this aims to be a general purpose function.

Reply:
I was just trying to see if I am missing a code that would require this. I am oik with this as it is more safe.